### ISSUE TITLE
[Tracker] Contact Method Flows Polish

### ISSUE THREAD

**Issue by Beamanator** [Expensify/App#16666]
Making this new issue to handle some follow-up tasks from https://github.com/Expensify/App/pull/15751

We merged that PR with a few unresolved follow-ups because I wanted to unblock @cristipaval 's work with the Add New Contact Method flow

There's a few follow-ups issues we need to investigate, will mention them here as they should not be considered "bugs" in Slack (in the near future I'll create issues for them)
- [x] ~We should definitely center the confirmation modal for removing a contact method~
    - I just pulled `main` and couldn't reproduce, so I think we can call this good to go?
- [x] Try to use the Form component for the Magic code input (& anywhere in the contact methods pages)
    - This should happen after we figure out exactly what the Offline UX should look like for that flow (trying to validate & resend magic code flows)
- [ ] Figure out best Offline UX flows for the 3 new flows recently added
    - Resend magic code
    - Validate a contact method
        - Currently there's not great feedback when offline
    - Delete a contact method
- [ ] [Bug](https://github.com/Expensify/App/pull/15751#issuecomment-1487705521): Following platforms do not submit the verify magic code from on press of the done/tick button on the Virtual keyboard.
    - iOS Native, iOS Safari, Android Native.
- [x] ~Passwordless - allow non-beta users to do the old password flows where needed :D~
    - Actually maybe not, since passwords may go away in the next few weeks!
- [x] Omit @expensify.sms everywhere possible ([example](https://github.com/Expensify/App/issues/16666#issuecomment-1488897967))
- [x] Limit magic code input to 6 digits (& only digits): https://expensify.slack.com/archives/C049HHMV9SM/p1680686551538709
- [x] https://github.com/Expensify/App/issues/17883
    - Always show default contact method on top of "Contact methods" list
- [ ] Don't show green checkmark immediately when resending magic code, only once the code has successfully sent (so if it fails, we don't see "yay I sent / oh shit i didn't send" ([reported here](https://expensify.slack.com/archives/C049HHMV9SM/p1680688202611159))
- [x] #20701
    - [Source](https://github.com/Expensify/App/pull/15330/files#r1158936999) - cc @cristipaval 
- [x] https://github.com/Expensify/App/issues/17722
- [ ] Setting a default contact method on two devices at the same time (similar [report](https://github.com/Expensify/App/pull/16750#issuecomment-1520601514)) (@cristipaval wants to work on this)
    - Basically instead of throwing "Already default contact method" error from the server, we should treat this as a "success" and send a pusher notification to the front-end as if this was a successful call (as long as this doesn't mess anything up in NewDot
- [x] https://expensify.slack.com/archives/C049HHMV9SM/p1682416824226729
- [ ] Think about this case: If I change my default contact method, do my contacts need to be immediately aware of this change? Will anything in their app break? Or can they refresh and fix everything?
    - Also need to handle this case on other of the user's devices (updating `personalDetails`)
- [x] https://github.com/Expensify/App/issues/17974
- [ ] Make RBR consistent with the rest of the app in Contact Methods: https://github.com/Expensify/App/issues/17827
    - [x] https://github.com/Expensify/App/issues/18252
- [ ] https://github.com/Expensify/App/issues/18575
- [x] https://expensify.slack.com/archives/C049HHMV9SM/p1684833850028879
- [x] https://github.com/Expensify/App/issues/19080
- [x] https://github.com/Expensify/Expensify/issues/286059
- [ ] Make new Auth command for "Add new contact method" so it's 1:1:1

**Comment by Beamanator** [comment]
Keeping this as "planning" for now, as I need to figure out the desired offline UX patterns (part 3) before completing part 2.

**Comment by NikkiWines** [comment]
Just dropping in here to say that we should probably omit the `@expensify.sms` here 

<img width="493" alt="image" src="https://user-images.githubusercontent.com/16747822/228599592-1415fa0c-dd85-422a-aeca-4c2ac2e285c2.png">

**Comment by Beamanator** [comment]
Still WIP - lots going on :D

**Comment by Beamanator** [comment]
Still in planning

**Comment by Beamanator** [comment]
Planning to keep this in "planning" until the "Set as default" App PR gets merged, as that's the final non-polish PR for Account Settings! (I believe)

**Comment by Beamanator** [comment]
Still working on getting the final "Set as default" App PR merged before moving forward on this list

**Comment by Beamanator** [comment]
@cristipaval assigning you here too since we're going to work on these polish issues together

**Comment by cristipaval** [comment]
Thank you!

**Comment by cristipaval** [comment]
Updated the title to help Melvin figure out that this one is on hold.

**Comment by Beamanator** [comment]
Slowly progressing, but focusing on bigger projects

**Comment by cristipaval** [comment]
I'm on [this one](https://github.com/Expensify/App/issues/20701), though it's not my top priority

**Comment by Beamanator** [comment]
Will try to start picking some of these up this week 👍

**Comment by cristipaval** [comment]
Checking off 
>[Bug](https://github.com/Expensify/App/pull/15751#issuecomment-1487705521): Following platforms do not submit the verify magic code from on press of the done/tick button on the Virtual keyboard.
iOS Native, iOS Safari, Android Native.

Not needed anymore since we now have the new input code component to validate a new contact method.

**Comment by cristipaval** [comment]
Checking off 
> Omit @expensify.sms everywhere possible (https://github.com/Expensify/App/issues/16666#issuecomment-1488897967)

It is fixed already.

**Comment by Beamanator** [comment]
still working through these slowly

**Comment by cristipaval** [comment]
been busy with wave4 stuff

**Comment by Beamanator** [comment]
Still pretty busy with wave4

**Comment by Beamanator** [comment]
Still paused cuz we're hyper focused on wave 4 & 5

**Comment by Beamanator** [comment]
Still paused cuz we're hyper focused on wave 4 & 5

**Comment by Beamanator** [comment]
Still paused cuz we're hyper focused on wave 4 & 5

**Comment by Beamanator** [comment]
we'll definitely get back to these after offshore 🙏

**Comment by cristipaval** [comment]
I would make this one Monthly, @Beamanator. I think detailed doc for the Onboarding flow has higher priority than this one.

**Comment by cristipaval** [comment]
thoughts about the above comment, @Beamanator ?

**Comment by Beamanator** [comment]
I agree! BTW you can make the change next time if you want :D

**Comment by Beamanator** [comment]
not prioritizing

**Comment by Beamanator** [comment]
same, not prioritizing

**Comment by cristipaval** [comment]
same

**Comment by cristipaval** [comment]
I'll look into this next week to see how many things still make sense.

**Comment by cristipaval** [comment]
I am going to close this one, given the urgency we have on waves. If anyone still thinks this has to be done, I think we have to go through the #whatsnext process
